src/main/java/com/commercetools/sync/producttypes/helpers/AttributeTypeAssert.java

```diff
@@ -0,0 +1,54 @@
+package com.commercetools.sync.producttypes.helpers;
+
+import com.commercetools.sync.commons.exceptions.BuildUpdateActionException;
+import io.sphere.sdk.products.attributes.AttributeType;
+
+public final class AttributeTypeAssert {
```

Comment:
I have multiple points against this class:

1. It is not very extensible. This is due to the existence of a lot code duplication that does the same thing. We should reuse this null checking logic. What if at some point we want to add more validations for an `attributeType` or an `attributeDefinition` (like checking the `name` also). Then we'd have to change all the three methods! (`assertTypesAreNull`, `assertOldAttributeType` and `assertOldAttributeType`).


2. It's also not very flexible to use. What if the caller wants to use these validation for other cases other than when building update actions? Why should it always throw a `BuildUpdateActionException` without knowing the actual use cases. These validation could actually be pretty useful to use even out of the case of building update actions.

3. I see the two methods (`assertOldAttributeType` and `assertOldAttributeType`) duplicated exactly. Just name changes of the method, paramater and throwable msg. We can definitely do better here and reuse.

4. No use of @Nullable annotations on params.
5. The error messages are not very informative. It's not helpful to the user. For example, when the user sees "Attribute type is not set for the old attribute definition." It's really hard to know where the error is. The attribute definition name would be great to have in this error message.
6. No unit tests :( :( :(


I would suggest to have this class be more of an internal validation util for `AttributeDefinition` and `AttributeDefinitionDraft`s (in the internal/util package) because why should we expose it. Something like:

````java
public final class AttributeDefinitionValidationUtils {
    public static boolean areValid(@Nonnull final AttributeDefinition oldAttributeDefinition,
                                   @Nonnull final AttributeDefinitionDraft newAttributeDefinitionDraft) {
        return isValid(oldAttributeDefinition) && isValid(newAttributeDefinitionDraft);
    }

    public static boolean isValid(@Nonnull final AttributeDefinition attributeDefinition) {
        return ofNullable(attributeDefinition.getAttributeType()).isPresent();
    }

    public static boolean isValid(@Nonnull final AttributeDefinitionDraft attributeDefinitionDraft) {
        return ofNullable(attributeDefinitionDraft.getAttributeType()).isPresent();
    }

    private AttributeDefinitionValidationUtils() {
    }
}
````

This way:
1. It becomes more extensible: If we decide to change the validation, we change in one method only isValid. And then this new validation logic would be automatically used in the other callers.
2. It becomes more flexible: the caller decides what to do in case it's invalid or valid. Rather than getting always a specific exception unrelated to validation. Another solution, as mentioned, would be to throw sth like `InvalidAttributeDefinition` and `InvalidAttributeDefinitionDraft` (but its more unnecessary work IMO)

Reply:
Thanks for the suggestions are good. I will implement them. Let me clear about my idea to create this class. First, I added this if statements in the methods but it was duplicated between in several places. So I have just moved them to a class. I have add test for this cases but I am not calling directly. So it should be internal like you said and the messages should be more clear. But in the first place, the idea was simple. I don't think, it should be extensible, flexible or exposed.

Reply:
https://github.com/commercetools/commercetools-sync-java/pull/316/commits/050df3b877dddd8f004ed05d403087a1ebad5754

Reply:
This commit reverts changes about attribute definition validations.
https://github.com/commercetools/commercetools-sync-java/pull/316/commits/8c3aa3cf627750feb717b5294b5072694fbb427f

Also, I made some utility methods package private to prevent validation errors. https://github.com/commercetools/commercetools-sync-java/pull/316/commits/925a829486748c5a8f1f964cebf3071f09cc6ad3 and 
https://github.com/commercetools/commercetools-sync-java/pull/316/commits/f213da6dac09b477a7a88d4d33a9f76910e96c77 . It's kind of breaking changes for product type but I'm not sure how we should explain it. Do you have an idea?